app/lib/admin/backend.dart

```diff
@@ -0,0 +1,200 @@
+  final DatastoreDB _db;
+  AdminBackend(this._db);
+
+  Future<R> _withAdmin<R>(Future<R> fn(AuthenticatedUser user)) async {
+    return await withAuthenticatedUser((u) async {
+      // TODO: remove once [withAuthenticatedUser] is using [User]
+      final user = await accountBackend.lookupUserById(u.userId);
+      if (!activeConfiguration.admins.contains(user.oauthUserId)) {
+        throw AuthorizationException.userIsNotAdminForPubSite();
+      }
+      return await fn(u);
+    });
+  }
+
+  /// List users.
+  Future<api.AdminListUsersResponse> listUsers(
+      {String continuationToken, int limit = 1000}) async {
+    InvalidInputException.checkRange(limit, 'limit', minimum: 1, maximum: 1000);
+    return await _withAdmin((user) async {
+      final query = _db.query<User>()
+        ..order('__key__')
+        ..limit(limit);
+
+      if (continuationToken != null) {
+        String lastId;
+        try {
+          lastId = _continuationCodec.decode(continuationToken);
+        } on FormatException catch (_) {
+          throw InvalidInputException.continuationParseError();
+        }
+        InvalidInputException.checkNotNull(lastId, 'continuationToken');
+
+        // NOTE: we should fix https://github.com/dart-lang/gcloud/issues/23
+        //       and remove the toDatastoreKey conversion here.
+        final key =
+            _db.modelDB.toDatastoreKey(_db.emptyKey.append(User, id: lastId));
+        if (lastId != null) {
+          query.filter('__key__ >', key);
+        }
+      }
+
+      final users = await query.run().toList();
+      // We may return a page with users less then a limit, but we always
+      // set the continuation token to the correct value.
+      final newContinuationToken = users.length < limit
+          ? null
+          : _continuationCodec.encode(users.last.userId);
+      users.removeWhere((u) => u.isDeleted);
+
+      return api.AdminListUsersResponse(
+        users: users
+            .map(
+              (u) => api.AdminUserEntry(
+                userId: u.userId,
+                email: u.email,
+                oauthUserId: u.oauthUserId,
+              ),
+            )
+            .toList(),
+        continuationToken: newContinuationToken,
+      );
+    });
+  }
+
+  /// Removes user from the Datastore and updates the packages and other
+  /// entities they may have controlled.
+  Future removeUser(String userId) async {
+    await _withAdmin((u) async {
+      // TODO: remove once [withAuthenticatedUser] is using [User]
+      final user = await accountBackend.lookupUserById(userId);
+      if (user == null) return;
+      if (user.isDeleted) return;
+
+      _logger.info(
+          '${u.userId} (${u.email}) initiated the delete of ${user.userId} (${user.email})');
+
+      // Package.uploaders
+      final pkgQuery = _db.query<Package>()..filter('uploaders =', user.userId);
+      await for (final p in pkgQuery.run()) {
+        await _removeUploaderFromPackage(p.key, user.userId);
```

Comment:
Maybe we should use: https://pub.dev/packages/pool
Such that we can do 10 concurrent calls...